aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,217 @@
+
+Forcing a Particular Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can apply the ``[Produces]`` filter. Like most :doc:`/mvc/controllers/filters`, this can be applied at the action, controller, or global scope. When you specify it, you provide the content type that should be produced as a string:
+
+.. code-block:: c#
+
+  [Produces("application/json")]
+  public class AuthorsController
+
+The above filter would force all actions within the AuthorsController to return JSON-formatted responses, even if other formatters were configured for the application and the client provided an ``Accept`` header requesting a different, available format.
+
+Special Case Formatters
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Some special cases are implemented using built-in formatters, as already described above. By default, ``string`` return types will be formatted as `text/plain`. This behavior can be removed by removing the ``TextPlainFormatter``. Likewise, actions that have a model object return type will return a 204 No Content resposne when returning ``null``. This behavior can be removed by removing the ``HttpNoContentOutputFormatter``. Removing formatters is done in *Startup* when configuring MVC.
+
+Response Format URL Mappings
+----------------------------
+
+Clients can request a particular format as part of the URL, such as in the querystring or part of the path, or by using a format-specific file extension (.xml, .json). The mapping from request path should be specified in the route the API is using. For example:
+
+.. code-block:: c#
+
+  [FormatFilter]
+  public class ProductsController
+  {
+  [Route("[controller]/[action]/[id].{format?}")]
+  public Product GetById(int id)
+  
+This route would allow the requested format to be specified as an optional file extension. The ``[FormatFilter]`` attribute checks for the existence of the format value in the ``RouteData`` and will map the output format to the appropriate formatter when the response is created.
+
+.. list-table:: Examples
+  :header-rows: 1
+  
+  * - Route
+    - Behavior
+  * - ``/products/GetById/5`` 
+    - Would use the default output format
+  * - ``/products/GetById/5.json``
+    - Would use the JSON formatter
+  * - ``/products/GetById/5.xml``
+    - Would use the XML formatter (if configured)
+
+Recommendations
+---------------
+
+There are many options when it comes to returning data from API actions. When creating demos or coding something quickly to see if it will work, simple return types may be appropriate. However, for APIs that will be used in production, you should prefer ``IActionResult`` as your return type for data since it allows you to handle error states and different HTTP status codes explicitly.
```

Comment:
>When creating demos or coding something quickly to see if it will work, simple return types may be appropriate.

Simple return types may be appropriate for prototype code.